### PR TITLE
PLG: use server-side models

### PR DESCRIPTION
**Pull request by abeatrix**
CLOSE https://linear.app/sourcegraph/issue/CODY-3800

- Add support to new fields added by https://github.com/sourcegraph/sourcegraph/pull/643
  - This ensures the server-side config for dotcom will be rendered as the client-side models currently we are currently using.
- Replace current client-side models with server-side models
- Remove DEFAULT_DOT_COM_MODELS and getDotComDefaultModels, or turn them into MOCK for testing
  - Separated into a different PR so that it's easier for review: https://github.com/sourcegraph/cody/pull/5725
- Fix an issue where we are using model ID with the current modelID without the provider, which is not recognized by the server. To fix this, i've updated all model id field to use the ModelRef when converted from the server-side mode config.

Here is the error that we get in current clients before the change:

image

## Test plan

Updated all the current tests to use the new config and added tests for supporting old configs.

Manual Testing: verify all the features are working as expected after the change.
- [ ] chat
- [ ] edit
- [ ] completion

Chat: 

confirm the models are showing up correctly in the dropdown:

image

confirm the request was submitted with the new model ref and that it works

image

Edit: Confirm the models are showing up in the model list and inline edit still work as expected1

image

image

Completions

image

## Changelog

Use server side model config

### CODE REVIEW

File: lib/shared/src/models/modelsService.ts [hunk 1]
```diff
@@ -37,16 +37,17 @@ export interface ModelRef {
...
     | ModelTag.Experimental
     | ModelTag.EarlyAccess
     | ModelTag.OnWaitlist
     | ModelTag.Waitlist
     | 'stable'
     | ModelTag.Deprecated
+    | ModelTag.Internal
 export type ModelTier = ModelTag.Free | ModelTag.Pro | ModelTag.Enterprise
-export type ModelCapability = 'chat' | 'autocomplete'
+export type ModelCapability = 'chat' | 'autocomplete' | 'edit' | 'streamDisabled' | 'vision'
```
Comment (olafurpg) [inline lib/shared/src/models/modelsService.ts:50]: woo hoo vision 🐙 :excited:

**Comment by abeatrix** [comment]
@jamesmcnamara the tests are failing after the last commit due to this error `Error: No default edit model found. Please set one.`
looks like it's having issue finding the default model for edit?

**Review by jamesmcnamara — APPROVED** [review]

**Comment by abeatrix** [comment]
@valerybugakov would like your eyes on the changes made to autocomplete 🙇‍♀️

File: lib/shared/src/models/model.ts [hunk 2]
```diff
@@ -180,3 +198,47 @@ export function parseModelRef(ref: ModelRefStr): ModelRef {
...
     }
 }
+
+/**
+ * Converts a model reference or ID to its legacy model ID format.
+ * If the input is a model reference, it extracts the modelId.
+ * If the input is already a legacy ID, it returns it unchanged.
+ *
+ * @param modelRefOrID - The model reference string or legacy model ID
+ * @returns The legacy model ID
```
Comment (valerybugakov) [inline lib/shared/src/models/model.ts:208]: Could you add an example here for clarity? E.g., `fireworks/starcoder-7b -> starcoder-7b`

**Review by valerybugakov — APPROVED** [review]
LGMT! TY for the ping

File: lib/shared/src/models/model.ts (hunk 2 above)
Comment (abeatrix) [reply lib/shared/src/models/model.ts:208]: Rip I just saw this. 
Will add it in a follow-up PR tomorrow 😃

**Pull request merged**